lib/rspec/core/example.rb

```diff
@@ -304,29 +304,55 @@ def inspect
 
       # @private
       #
-      # Used internally to set an exception in an after hook, which
-      # captures the exception but doesn't raise it.
-      def set_exception(exception, context=nil)
-        if pending? && !(Pending::PendingExampleFixedError === exception)
-          execution_result.pending_exception = exception
+      # The exception that will be displayed to the user -- either the failure of
+      # the example or the `pending_exception` if the example is pending.
+      def display_exception
+        @exception || execution_result.pending_exception
+      end
+
+      # @private
+      #
+      # Assigns the exception that will be displayed to the user -- either the failure of
+      # the example or the `pending_exception` if the example is pending.
+      def display_exception=(ex)
+        if pending? && !(Pending::PendingExampleFixedError === ex)
+          @exception = nil
+          execution_result.pending_fixed = false
+          execution_result.pending_exception = ex
         else
-          if @exception
-            # An error has already been set; we don't want to override it,
-            # but we also don't want silence the error, so let's print it.
-            msg = <<-EOS
+          @exception = ex
+        end
+      end
 
-  An error occurred #{context}
-    #{exception.class}: #{exception.message}
-    occurred at #{exception.backtrace.first}
+      # rubocop:disable Style/AccessorMethodName
 
-            EOS
-            RSpec.configuration.reporter.message(msg)
-          end
+      # @private
+      #
+      # Used internally to set an exception in an after hook, which
+      # captures the exception but doesn't raise it.
+      def set_exception(exception)
+        return self.display_exception = exception unless display_exception
 
-          @exception ||= exception
+        unless RSpec::Core::MultipleExceptionError === display_exception
+          self.display_exception = RSpec::Core::MultipleExceptionError.new(display_exception)
         end
+
+        display_exception.add exception
+      end
+
+      # @private
+      #
+      # Used to set the exception when `aggregate_failures` fails.
+      def set_aggregate_failures_exception(exception)
```

Comment:
The rubocop exception was added after-the-fact because it complained.  Mostly I was mirroring `set_exception` which used to take 2 arguments (and therefore could not be `exception=`).  In my experience, it's common to have logical reader methods but far less common to have logical writer methods, and having `aggregate_failures_exception=` suggests (to me at least) that there would be a corresponding `aggregate_failures_exception` reader method and there's not.  Really, this isn't modeling an attribute; it's performing the necessary logic when an aggregate failure happens in the `around` hook.